feat: Add Datadog as an exporter for OTLP traces (#2083)
flipt-io/flipt: examples/tracing/otlp/README.md

1. Select `serviceName=flipt` from the search box
1. Click 'Run Query'
1. You should see a list of traces to explore

### Datadog UI

!['Datadog Example'](../../images/datadog.png)

> For exporting traces from OpenTelemetry to Datadog you have to provide this piece of configuration in the `otel-collector-config.yaml`:

Reviewer: ```suggestion
For exporting traces from [OpenTelemetry to Datadog](https://docs.datadoghq.com/opentelemetry/otel_collector_datadog_exporter) you have to configure the exporter in the `otel-collector-config.yaml`:
```

```yaml
exporters:
  datadog:
    api:
      site: datadoghq.com
      key: ${DD_API_KEY}
```

**Note:** The `DD_API_KEY` should be replaced with your actual api key from Datadog.

> You can do this under the extensions configuration. You also have to add `datadog` as an entry in `exporters` under `service.pipelines.traces.exporters`.

Reviewer: do what under the extensions configuration?

Author: @markphelps Simplified the wording on this.

> You can do this under the extensions configuration. You also have to add `datadog` as an entry in `exporters` under `service.pipelines.traces.exporters`.

Reviewer: You can do this under the extensions configuration. You also have to add `datadog` as an entry in `exporters` under `service.pipelines.traces.exporters`. 

For example:

```
service:
  extensions: [pprof, zpages, health_check]
  pipelines:
    traces:
      receivers: [otlp]
      processors: [batch]
      exporters: [logging, datadog]
```
